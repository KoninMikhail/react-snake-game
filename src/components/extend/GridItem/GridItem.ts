import styled from 'styled-components';

/**
 * @typedef IGridItemProps
 *
 * @property {string=} gridColumn - Columns size (css: grid-column);
 * @property {string=} gridRow - Row size (css: grid-row);
 * @property {string=} align - Align in grid item (css: justify-self);
 * @property {string=} justify- Justify in grid item (css: align-self);
 */

interface IGridItemProps {
  gridColumn?: string;
  gridRow?: string;
  justify?: string;
  align?: string;
}

/**
 * GridItem
 *
 * @description Use for creation item in the grid
 *
 * @memberOf Grid
 * @param {IGridItemProps} - The grid item properties
 * @returns {JSX.Element}
 */

export const GridItem = styled.div<IGridItemProps>`
  display: block;
  position: relative;
  grid-column: ${(props) => props?.gridColumn};
  grid-row: ${(props) => props?.gridRow || 'auto'};
  justify-self: ${(props) => props?.justify || 'auto'};
  align-self: ${(props) => props?.align || 'auto'};
`;
