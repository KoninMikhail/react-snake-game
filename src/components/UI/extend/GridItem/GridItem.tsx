import { CSSProperties } from 'styled-components';
import { StyledGridItem } from './StyledGridItem';

/**
 * @typedef IGridItemProps
 *
 * @property {string=} gridColumn - Columns size (css: grid-column);
 * @property {string=} gridRow - Row size (css: grid-row);
 * @property {string=} align - Align in grid item (css: justify-self);
 * @property {string=} justify- Justify in grid item (css: align-self);
 */

export interface IGridItemProps {
  gridColumn?: string;
  gridRow?: string;
  justify?: string;
  align?: string;
  children: JSX.Element | JSX.Element[] | string;
  style?: CSSProperties;
}

/**
 * GridItem
 *
 * Use for creation item in the grid
 *
 * @memberOf Grid
 * @param {IGridItemProps} props - The grid item properties
 * @returns {JSX.Element}
 */

export const GridItem = ({ children, ...rest }: IGridItemProps): JSX.Element => (
  <StyledGridItem {...rest}>{children}</StyledGridItem>
);
