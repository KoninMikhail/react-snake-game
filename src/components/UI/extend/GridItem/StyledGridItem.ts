import styled from 'styled-components';
import { IGridItemProps } from './GridItem';

export const StyledGridItem = styled.div<IGridItemProps>`
  display: block;
  position: relative;
  grid-column: ${(props) => props?.gridColumn || 'auto'};
  grid-row: ${(props) => props?.gridRow || 'auto'};
  justify-self: ${(props) => props?.justify || 'auto'};
  align-self: ${(props) => props?.align || 'auto'};
`;
