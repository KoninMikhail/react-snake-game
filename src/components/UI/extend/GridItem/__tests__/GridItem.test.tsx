import { render, screen } from '@testing-library/react';
import { GridItem } from '../GridItem';

describe('GridItem', () => {
  const gridItemProps = {
    justify: 'center',
    align: 'center',
    gridColumn: 'main-start',
    gridRow: 'span 3',
  };
  const childrenContent = 'childrenContent';
  const gridItemTestId = 'gridItemTestId';

  it('should be render with props', () => {
    render(
      <GridItem data-testid={gridItemTestId} {...gridItemProps}>
        {childrenContent}
      </GridItem>
    );

    const itemContainer = screen.getByTestId(gridItemTestId);

    expect(itemContainer).toHaveStyleRule('display', 'block');
    expect(itemContainer).toHaveStyleRule('grid-column', 'main-start');
    expect(itemContainer).toHaveStyleRule('grid-row', 'span 3');
    expect(itemContainer).toHaveStyleRule('justify-self', 'center');
    expect(itemContainer).toHaveStyleRule('align-self', 'center');
  });

  it('should be render without props', () => {
    render(<GridItem data-testid={gridItemTestId}>{childrenContent}</GridItem>);

    const itemContainer = screen.getByTestId(gridItemTestId);

    expect(itemContainer).toHaveStyleRule('display', 'block');
    expect(itemContainer).toHaveStyleRule('grid-column', 'auto');
    expect(itemContainer).toHaveStyleRule('grid-row', 'auto');
    expect(itemContainer).toHaveStyleRule('justify-self', 'auto');
    expect(itemContainer).toHaveStyleRule('align-self', 'auto');
  });

  it('equal snapshot', () => {
    render(<GridItem data-testid={gridItemTestId}>{childrenContent}</GridItem>);
    expect(screen.getByTestId(gridItemTestId)).toMatchSnapshot();
  });
});
