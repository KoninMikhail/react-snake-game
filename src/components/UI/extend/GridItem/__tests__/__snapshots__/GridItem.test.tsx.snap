// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridItem equal snapshot 1`] = `
.c0 {
  display: block;
  position: relative;
  grid-column: auto;
  grid-row: auto;
  justify-self: auto;
  -webkit-align-self: auto;
  -ms-flex-item-align: auto;
  align-self: auto;
}

<div
  class="c0"
  data-testid="gridItemTestId"
>
  childrenContent
</div>
`;
